front/lib/api/assistant/agent.ts

```diff
@@ -202,20 +203,42 @@ async function* runMultiActionsAgentLoop(
           // against the model outputting something unreasonable.
           event.actions = event.actions.slice(0, MAX_ACTIONS_PER_STEP);
 
-          const eventStreamGenerators = event.actions.map(
-            ({ action, inputs, functionCallId, specification }, index) => {
+          // We store the actions in the database using bulkCreate for efficiency.
+          const plannedActions = await AgentMessageStepPlanning.bulkCreate(
+            event.actions.map((action, stepActionIndex) => ({
+              workspaceId: auth.getNonNullableWorkspace().id,
+              agentMessageId: agentMessage.agentMessageId,
+              step: i,
+              stepActionIndex,
+              actionConfiguration: action.action,
```

Comment:
Why do we need to store this one? This is a full AgentActionConfiguration which has been pulled and renderded by `runMultiActionsAgent`.

I thiiink what we'd like to store is the ~raw output of the model `output.actions[i]` on l.747?

We can delay finding the associated action and running it in the later activities triggered from these?

Reply:
I'm refactoring that! 👍🏻